src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -115,39 +116,48 @@ trait WorkflowComponent {
+        attributeNamespaceQuery.getMap flatMap { attributeNamespaceMapping =>
 
-        val insertedRecQuery = for {
-          workflowRec <- findWorkflowsBySubmissionId(submissionId)
-          workflowEntityRec <- entityQuery if workflowEntityRec.id === workflowRec.workflowEntityId
-          insertedInputResolutionRec <- submissionValidationQuery if insertedInputResolutionRec.workflowId === workflowRec.id
-        } yield (workflowEntityRec, insertedInputResolutionRec)
+          val inputResolutionRecs: Seq[SubmissionValidationRecord] = for {
+            workflow <- workflows
+            inputResolution <- workflow.inputResolutions
+          } yield marshalInputResolution(inputResolution, attributeNamespaceMapping(inputResolution.inputName.namespace), workflowRecsByEntity(workflow.workflowEntity).id)
+
+          val insertedRecQuery = for {
+            workflowRec <- findWorkflowsBySubmissionId(submissionId)
+            workflowEntityRec <- entityQuery if workflowEntityRec.id === workflowRec.workflowEntityId
+            insertedInputResolutionRec <- submissionValidationQuery if insertedInputResolutionRec.workflowId === workflowRec.id
+          } yield (workflowEntityRec, insertedInputResolutionRec)
 
-        insertInBatches(submissionValidationQuery, inputResolutionRecs) andThen
-        insertedRecQuery.result.map(_.map { case (workflowEntityRec, insertedInputResolutionRec) =>
-          (AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name), insertedInputResolutionRec.inputName) -> insertedInputResolutionRec
-        }.toMap)
+          insertInBatches(submissionValidationQuery, inputResolutionRecs) andThen
+            insertedRecQuery.result.map(_.map { case (workflowEntityRec, insertedInputResolutionRec) =>
+              (AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name), DefaultAttributeName(insertedInputResolutionRec.inputName)) -> insertedInputResolutionRec
+            }.toMap)
+        }
       }
 
-      def insertInputResolutionAttributes(workflows: Seq[Workflow], inputResolutionRecs: Map[(AttributeEntityReference, String), SubmissionValidationRecord]) = {
-        val attributeRecs = for {
-          workflow <- workflows
-          inputResolution <- workflow.inputResolutions
-          attribute <- inputResolution.value
-        } yield submissionAttributeQuery.marshalAttribute(inputResolutionRecs(workflow.workflowEntity, inputResolution.inputName).id, inputResolution.inputName, attribute, Map.empty)
+      def insertInputResolutionAttributes(workflows: Seq[Workflow], inputResolutionRecs: Map[(AttributeEntityReference, AttributeName), SubmissionValidationRecord]): ReadWriteAction[Unit] = {
+        attributeNamespaceQuery.getMap flatMap { attributeNamespaceMapping =>
+          val attrTuples: Seq[(Workflow, SubmissionValidationValue, Attribute)] = for {
```

Comment:
Split up the original `for` because it was creating `Seq[Seq[]]` and flattening them.